src/Test/TestCommanderTech.java

```diff
@@ -0,0 +1,122 @@
+package Test;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import CommandingOfficers.Tech;
+import CommandingOfficers.Commander;
+import CommandingOfficers.CommanderAbility;
+import CommandingOfficers.Patch;
+import Engine.GameScenario;
+import Engine.XYCoord;
+import Engine.Combat.DamagePopup;
+import Engine.GameEvents.GameEventQueue;
+import Terrain.MapLibrary;
+import Terrain.MapMaster;
+
+public class TestCommanderTech extends TestCase
+{
+  private Commander mPatch;
+  private Tech mTech;
+  private MapMaster mTestMap;
+
+  private void setupTest()
+  {
+    GameScenario scn = new GameScenario();
+    mTech = new Tech(scn.rules);
+    mTech.modifyAbilityPower(20);
+    mPatch = new Patch(scn.rules);
+    Commander[] cos = { mTech, mPatch };
+
+    mTestMap = new MapMaster(cos, MapLibrary.getByName("Test Range"));
+  }
+
+  private void cleanupTest()
+  {
+    mTech = null;
+    mPatch = null;
+    mTestMap = null;
+  }
+
+  private CommanderAbility findAbilityByName(Commander co, String name)
+  {
+    CommanderAbility pow = null;
+
+    ArrayList<CommanderAbility> abilities = mTech.getReadyAbilities();
+    for( CommanderAbility ca : abilities )
+    {
+      if( ca.toString().equalsIgnoreCase(name) )
+      {
+        pow = ca;
+        break;
+      }
+    }
+
+    return pow;
+  }
+
+  @Override
+  public boolean runTest()
+  {
+    boolean testPassed = true;
+
+    testPassed &= validate(testTechDropRandomization(), "  Tech Drops are random!");
+
+    return testPassed;
+  }
+
+  private boolean testTechDropRandomization()
+  {
+    boolean testPassed = true;
+
+    // Let's make sure the predictions and generated events always match.
+    for( int i = 0; i < 10; i++ )
```

Comment:
Seems a little silly to run the calc 10 times with the same initial conditions, but whatever boats your float?

Reply:
Well, yeah. I initially built the test case under the assumption that the results were non-deterministic, and figured ten identical triple-drops was enough to pretty well guarantee it truly was deterministic.

Turns out it was deterministic the whole time, and just didn't agree with the damage previews. Guess I could simplify now, since the test always failed on the first pass before anyway.